tests/util/test_executable_runner.py

```diff
@@ -29,22 +30,32 @@ def test_validate_executable_path_not_executable() -> None:
             ExecutableRunner.validate_executable_path(executable=tmpfile.name)
 
 
-def test_validate_executable_path() -> None:
-    exec = "yes"
-    exec_full_str = f"/usr/bin/{exec}"
-    exec_full_path = Path(exec_full_str)
-    assert exec_full_path.is_absolute()
+@pytest.mark.parametrize("executable", ["yes", "/usr/bin/yes", Path("/usr/bin/yes")])
+def test_validate_executable_path(executable: str | Path) -> None:
+    """
+    `validate_executable_path` should find the `yes` executable in the following scenarios:
+    1. when the string "yes" is passed
+    2. when the absolute path to the `yes` executable is passed, either as a string or a Path
+    """
+    expected_path: Path = Path("/usr/bin/yes")
 
-    # find it on the PATH
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec)
-    # find it given an absolute path as a string
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec_full_str)
-    # find it given an absolute path as a Path
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec_full_path)
+    with mock.patch.dict(os.environ):
```

Comment:
NB: I'm using `unittest.mock` instead of `pytest.mocker` because of a bug/shortcoming in `pytest`. 

`patch.dict` only works as a context manager with `os.environ` if you use the standard library, it's not supported in `pytest`'s implementation

Reply:
OOI do you see it as better/worse/indifferent to go the `clear=True` route and just set the `PATH` in `os.environ`? 

It feels like it'd be a cleaner test to do it that way but a) I can see arguments against it and b) perhaps there's a reason I didn't see why that's not really feasible

Reply:
Ok I just updated to that effect. Dunno what blocked me there yesterday 🙃